compiler/optimizer/src/main/java/org/apache/nemo/compiler/optimizer/pass/compiletime/reshaping/SkewReshapingPass.java

```diff
@@ -47,17 +41,13 @@
 
 /**
  * Pass to reshape the IR DAG for skew handling.
- *
- * This pass inserts vertices to perform two-step dynamic optimization for skew handling.
- * 1) Task-level statistic collection is done via vertex with {@link MetricCollectTransform}
- * 2) Stage-level statistic aggregation is done via vertex with {@link AggregateMetricTransform}
- * inserted before shuffle edges.
+ * We insert a {@link MessageBarrierVertex for each shuffle edge}
```

Comment:
We **must** optimize multiple shuffle edges at once if a vertex receives many shuffle edges. Please check [327](https://issues.apache.org/jira/projects/NEMO/issues/NEMO-327?filter=allopenissues).